nshlib/nsh_prompt.c

```diff
@@ -0,0 +1,106 @@
+ * apps/nshlib/nsh_prompt.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <termios.h>
+
+#ifdef CONFIG_NSH_CLE
+#  include "system/cle.h"
+#else
+#  include "system/readline.h"
+#endif
+
+#include "nsh.h"
+#include "nsh_console.h"
+
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+
+static char g_nshprompt[CONFIG_NSH_PROMPT_MAX] = CONFIG_NSH_PROMPT_STRING;
+
+static const char g_prompt_suffix[] = CONFIG_NSH_PROMPT_SUFFIX;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nsh_refresh_prompt
+ *
+ * Description:
+ *   This function updates g_nshprompt with latest data based on prioritized
+ *   sources, from highest to lowest priority:
+ *
+ *   - non-empty NSH_PROMPT_STRING w/o appending suffix
+ *   - non-empty NSH_PROMPT_ENV variable value with suffix appended
+ *   - non-empty HOSTNAME with g_prompt_suffix appended
+ *   - g_prompt_suffix
+ *
+ * Results:
+ *   - updated g_nsh_prompt value if necessary.
+ *
+ ****************************************************************************/
+
+void nsh_update_prompt(void)
+{
+  if (CONFIG_NSH_PROMPT_STRING[0] == '\0')
+    {
+#ifndef CONFIG_DISABLE_ENVIRON
+      if (getenv(CONFIG_NSH_PROMPT_ENV))
+        {
+          strlcpy(g_nshprompt, getenv(CONFIG_NSH_PROMPT_ENV),
+                  sizeof(g_nshprompt));
+        }
+      else
+#endif
+        {
+          gethostname(g_nshprompt, sizeof(g_nshprompt));
+        }
+
+      uint16_t j = strlen(g_nshprompt) + strlen(g_prompt_suffix) + 1 ;
```

Comment:
1.change uint16_t to size_t
2.move the declaration to the begin of the block
3.add macro for sizeof(g_prompt_suffix)